---
title: "Fig_4_Viz"
author: "Troy McDiarmid"
date: "2023-10-30"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(ggrepel)
library(UpSetR)
library(ComplexHeatmap)
library(Gviz)
library(GenomicInteractions)
library(GenomicRanges)
library(InteractionSet)
library(tidyverse)
library(biomaRt)
library(readxl)
library(sjmisc)
library(Matrix)
library(sceptre)
library(VGAM)
library(hdf5r)
library(Seurat)
library(cowplot)
library(ggridges)
```

```{r}

##Read in hits

Hits <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Hits_Annotated.csv")

##Read in processed gRNA and GEX FBC matrices
  
Gene_Matrix <- readRDS("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/SCEPTRE_Processed_Matrices/Scaled_Screen_gene_Matrix.mtx")
gRNA_Matrix <- readRDS("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/SCEPTRE_Processed_Matrices/Scaled_Screen_gRNA_Matrix.mtx")

```


```{r}

##Fig. S4b

set.seed(2)

##ZFHX4 boxplots

##Defining gRNA & Target gene

gRNA <- "8849_chr8:75682018-75682941()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_8849_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)

##ZFHX4 gRNA #2

##Defining gRNA & Target gene

gRNA <- "8852_chr8:76525991-76526491()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_8852_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)

##ZFHX4 gRNA #3

##Defining gRNA & Target gene

gRNA <- "8859_chr8:76677643-76678426()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_8859_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)

##ZFHX4 gRNA #4

##Defining gRNA & Target gene

gRNA <- "8863_chr8:76683646-76684156()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_8863_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)


##ZFHX4 gRNA #5

##Defining gRNA & Target gene

gRNA <- "12689_ZFHX4;ENST00000651372::chr8:76680746-76681246(+)_Promoter_Flashfry"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#E69F00")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_12689_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)

##ZFHX4 gRNA #6

##Defining gRNA & Target gene

gRNA <- "14111_ZFHX4_Promoter_Horlbeck_etal"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#E69F00")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_14111_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)

##ZFHX4 gRNA #7

##Defining gRNA & Target gene

gRNA <- "14112_ZFHX4_Promoter_Horlbeck_etal"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#E69F00")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_14112_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)

##ZFHX4 gRNA #8

##Defining gRNA & Target gene

gRNA <- "8865_chr8:76683646-76684156()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000091656"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_8865_ZFHX4_Boxplot.jpeg", width = 3.75, height = 4.75)

```

```{r}
#Fig. S4b

##ZFHX4 Gviz

##Gviz plot 

# Set genomic coordinates for plotting
symbol = "ZFHX4"
genome = "hg38"
chrom = "chr8"
chromStart = 75600000
chromEnd = 77000000

# load biomaRt data from ensembl
biomart = biomaRt::useMart(biomart = "ensembl", dataset = "hsapiens_gene_ensembl")

# create an ideogram track to display chromosome
#iTrack <- IdeogramTrack(genome = genome, chromosome = chrom, name = chrom)

# create a genome axis track to display chromosome coordinates
gTrack <- GenomeAxisTrack()

# create track with ensembl genes
biomTrack <- BiomartGeneRegionTrack(genome = genome, chromosome = chrom, name = "ENSEMBL", 
                                    filter = list(with_refseq_mrna=TRUE), biomart = biomart)



### (2) Create plot of candidate enhancers, only show contacts for CRT gene list #############################

# Get the list of CRT candidate genes within the genomic window
CRISPRa_CRT_candidates <- data.frame(read_xlsx(path = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/Table_S1_Cis_Regulation_Therapy_Candidate_Genes.xlsx", sheet = "Candidate_Genes"))
CRISPRa_CRT_candidates$gene_symbol[which(CRISPRa_CRT_candidates$chrom == chrom & CRISPRa_CRT_candidates$start_hg38 > chromStart & CRISPRa_CRT_candidates$end_hg38 < chromEnd)]

# Get all 5,425 candidate enhancers of CRT genes
candidate_enhancers_for_CRT_genes <- read.table(file = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/candidate_enhancer_regions_for_CRISPRa_CRT_genes.bed", header = FALSE, sep = "\t",
                                                col.names = c("chrom", "chromStart", "chromEnd", "name"))

# Filter the genomic window of candidate enhancers for plotting
candidate_enhancers_for_CRT_genes <- candidate_enhancers_for_CRT_genes[which(candidate_enhancers_for_CRT_genes$chrom == chrom & candidate_enhancers_for_CRT_genes$chromStart > chromStart & candidate_enhancers_for_CRT_genes$chromEnd < chromEnd),]

# Create regulatory element annotation track
REtrack <- AnnotationTrack(start = candidate_enhancers_for_CRT_genes$chromStart, 
                           end = candidate_enhancers_for_CRT_genes$chromEnd, 
                           chromosome = chrom, genome = genome, stacking = "dense",
                           name = "Candidate Regulatory Elements") 


# Get all 761 TSSs of CRT genes
TSSs_for_CRT_genes <- read.table(file = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/CRISPRa_CRT_Candidates_gencode.v38.basic.coding.transcripts.500bp_promoters.bed", header = FALSE, sep = "\t",
                                                col.names = c("chrom", "chromStart", "chromEnd", "name", "number", "strand")) %>% 
  type_convert()

# Filter the genomic window of candidate enhancers for plotting
TSSs_for_CRT_genes <- TSSs_for_CRT_genes[which(TSSs_for_CRT_genes$chrom == chrom & TSSs_for_CRT_genes$chromStart > chromStart & TSSs_for_CRT_genes$chromEnd < chromEnd),]

# Create TSS annotation track
TSStrack <- AnnotationTrack(start = TSSs_for_CRT_genes$chromStart, 
                           end = TSSs_for_CRT_genes$chromEnd, 
                           chromosome = chrom, genome = genome, stacking = "dense",
                           name = "Transcription_Start_Sites") 


# Create genomic interaction dataframe for each candidate enhancer dataset
GW18_PFC_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/gw18_pfc_ABC_Score_liftOver_hg38_NDD.bedpe",
                                                type = "bedpe", experiment_name = "GW18_PFC", description = "ABC_Score")
WTC11_iNeuron_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/NGN2-neuron_ABC_Score_hg38_MPRA_CRISPRa_validated_for_Gviz.bedpe",
                                                     type = "bedpe", experiment_name = "WTC11_iNeuron", description = "ABC_Score")
Cicero_Fetal_Cerebrum_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/midfetal_cerebrum_ExN_Cicero_liftOver_hg38_NDD.bedpe",
                                                             type = "bedpe", experiment_name = "Cicero_Fetal_Cerebrum", description = "Cicero")
Trevino_Fetal_Cortex_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/trevino_fetal_cortex_enhancer_predictions_NDD.bedpe",
                                                            type = "bedpe", experiment_name = "Trevino_Fetal_Cortex", description = "Correlation")
Ziffra_Fetal_Cortex_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/ziffra_fetal_human_cortex_ABC_Score_NDD_for_gviz_figures_only.bedpe",
                                                           type = "bedpe", experiment_name = "Ziffra_Fetal_Cortex", description = "ABC_Score")

# only plot positive scores for Trevino
Trevino_Fetal_Cortex_NDD <- Trevino_Fetal_Cortex_NDD[which(Trevino_Fetal_Cortex_NDD@elementMetadata$counts > 0),]

# the height of the interaction plot represents the score from each prioritization method
summary(GW18_PFC_NDD@elementMetadata$counts)
summary(WTC11_iNeuron_NDD@elementMetadata$counts)
summary(Cicero_Fetal_Cerebrum_NDD@elementMetadata$counts)
summary(Trevino_Fetal_Cortex_NDD@elementMetadata$counts)
summary(Ziffra_Fetal_Cortex_NDD@elementMetadata$counts)

# Create interaction tracks for each candidate enhancer dataset
GW18_PFC_NDD_interaction_track <- InteractionTrack(GW18_PFC_NDD, name = "GW18_PFC", chromosome = chrom)
WTC11_iNeuron_NDD_interaction_track <- InteractionTrack(WTC11_iNeuron_NDD, name = "WTC11_iNeuron", chromosome = chrom)
Cicero_Fetal_Cerebrum_NDD_interaction_track <- InteractionTrack(Cicero_Fetal_Cerebrum_NDD, name = "Cicero_Fetal_Cerebrum", chromosome = chrom)
Trevino_Fetal_Cortex_NDD_interaction_track <- InteractionTrack(Trevino_Fetal_Cortex_NDD, name = "Trevino_Fetal_Cortex", chromosome = chrom)
Ziffra_Fetal_Cortex_NDD_interaction_track <- InteractionTrack(Ziffra_Fetal_Cortex_NDD, name = "Ziffra_Fetal_Cortex", chromosome = chrom)

# Show the available display parameters for a Gviz track
# availableDisplayPars(gTrack)


##Adding an ENCODE ATAC track


bamFile <- "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/WTC11_NGN2_iPSC_7-8wk_ExN_ATAC-seq_Song_2019_hg38.50bp_trim.merged.srt.nodup.no_chrM_MT.filtered.bw"
ATAC_track <- DataTrack(range = bamFile, genome = genome, type = "l", 
                     name = "ATAC", window = -1, 
                     chromosome = chrom, type = "hist", 
                    col.histogram = "black", 
                    fill.histogram = "black", background.title = "white", 
                    fontcolor.title = "black", ylim = c(0,5))
class(ATAC_track)

ATAC_track



##Adding a H3k27Ac track


bamFile <- "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/WTC11_NGN2_iPSC_7-8wk_ExN_H3K27ac_CUT+RUN_Song_2019_hg38.srt.no_dups.filtered.bw"
H3k27ac <- DataTrack(range = bamFile, genome = genome, type = "l", 
                     name = "H3K27AC", window = -1, 
                     chromosome = chrom, type = "hist", 
                    col.histogram = "black", 
                    fill.histogram = "black", background.title = "white", 
                    fontcolor.title = "black", ylim = c(0,10))
class(H3k27ac)

H3k27ac


##Then reading in the gRNA mappings and function data and generating the gRNA function score track

gRNA_Coordinates <- read_tsv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/Scaled_Screen_gRNA_coordinates.txt") %>% 
  dplyr::rename(Chromosome = seqname, gRNA_Name = patternID) %>% 
  separate(Chromosome, into = c("Chr", "Chromosome"),
            sep= 3) %>%
  arrange(gRNA_Name) 

gRNA_Coordinates <- gRNA_Coordinates %>% 
  separate(gRNA_Name, into = c("gRNA_ID", "gRNA_Name"), sep = "_", remove = FALSE) %>% 
  type_convert()

gRNA_DE_Results <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Results.csv") 

gRNA_DE_Results <- gRNA_DE_Results %>% 
  left_join(gRNA_Coordinates, by = "gRNA_ID")

##convert ensg ids to hgnc symbols and filter

mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- gRNA_DE_Results$response_id
G_list <- getBM(filters= "ensembl_gene_id", attributes= c("hgnc_symbol", "ensembl_gene_id"), values=genes, mart= mart)

G_list <- G_list %>% 
  dplyr::rename(response_id = ensembl_gene_id) %>% 
  dplyr::rename(target_gene = hgnc_symbol)

gRNA_DE_Results <- gRNA_DE_Results %>% 
  left_join(G_list, by = "response_id")

gRNA_Data <- gRNA_DE_Results %>% 
  filter(target_gene == symbol) %>% 
  unite(gRNA_Gene_Pair, gRNA_ID, target_gene, remove = FALSE) 

#gRNA_Data <- gRNA_Data %>%
  #distinct(gRNA_Gene_Pair, .keep_all = TRUE)

gRNA_Data <- gRNA_Data %>%
  mutate(plot_pval = -log10(p_value))

##Filtering for hits or non-hits and coloring 

PT_Hits <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Hits_Annotated.csv")

Promoter_Hits <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Hits_Annotated.csv") %>% filter(Target_CRE_Class == "Promoter")

Enhancer_Hits <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Hits_Annotated.csv") %>% filter(Target_CRE_Class == "Enhancer")

Promoter_Hits <- gRNA_Data %>% 
  filter(gRNA_ID %in% Promoter_Hits$gRNA_ID)
Promoter_Hits$Hit <- "Hit"

Enhancer_Hits <- gRNA_Data %>% 
  filter(gRNA_ID %in% Enhancer_Hits$gRNA_ID)
Enhancer_Hits$Hit <- "Hit"

Non_Hits <- gRNA_Data %>% 
  filter(!gRNA_ID %in% PT_Hits$gRNA_ID)
Non_Hits$Hit <- "Non-Hit" 

##Create hit and non-hit tracks and overlay them

df1_label = "Non-Hit"
df2_label = "Promoter_Hit"
df3_label = "Enhancer_Hit"
df1 <- Non_Hits
df2 <- Promoter_Hits
df3 <- Enhancer_Hits

Enhancer_Hit_Track <- DataTrack(data = Enhancer_Hits$plot_pval, start = Enhancer_Hits$start, end = Enhancer_Hits$end, chromosome = chrom, genome = genome, groups = factor(df3_label,levels = c(df1_label, df2_label, df3_label)), col = "#56B4E9", cex = 1.2, legend=FALSE, ylim = c(0,14), col.axis = "black", background = "white", fontcolor.title = "black", name = "-log10(P-values)", background.title = "white", col.frame = "white", fill = "white")

Promoter_Hit_Track <- DataTrack(data = Promoter_Hits$plot_pval, start = Promoter_Hits$start, end = Promoter_Hits$end, chromosome = chrom, genome = genome, groups = factor(df2_label,levels = c(df1_label, df2_label, df3_label)), col = "#E69F00", cex = 1.2, legend=FALSE, ylim = c(0,14), col.axis = "black", background = "white", fontcolor.title = "black", name = "-log10(P-values)", background.title = "white", col.frame = "white", fill = "white")

Non_Hit_Track <- DataTrack(data = Non_Hits$plot_pval, start = Non_Hits$start, end = Non_Hits$end, chromosome = chrom, genome = genome, groups = factor(df1_label,levels = c(df1_label, df2_label)), col = "#999999", legend=FALSE, ylim = c(0,14), col.axis = "black", background = "white", fontcolor.title = "black", name = "-log10(P-values)", background.title = "white", col.frame = "white", fill = "white")

overlapTracks <- OverlayTrack(trackList = list(Non_Hit_Track, Promoter_Hit_Track, Enhancer_Hit_Track), legend= TRUE, background.title = "white", col.frame = "white", fill = "white")



# Set track parameters for plotting
displayPars(gTrack) <- list(fontsize = 6)
displayPars(GW18_PFC_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(WTC11_iNeuron_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(Cicero_Fetal_Cerebrum_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(Trevino_Fetal_Cortex_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(Ziffra_Fetal_Cortex_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(REtrack) <- list(col = "#56B4E9", fill = "#56B4E9", col.axis = "black", background.title = "white", fontcolor.title = "black")
displayPars(TSStrack) <- list(col = "#E69F00", fill = "#E69F00", col.axis = "black", background.title = "white", fontcolor.title = "black")
displayPars(biomTrack) <- list(background.title = "white", fontcolor.title = "black", col = "#999999", fill = "#999999", col.line = "#999999", utr3 = "#999999", utr5 = "#999999", protein_coding = "#999999")
displayPars(ATAC_track) <- list(background.title = "white", fontcolor.title = "black")
displayPars(H3k27ac) <- list(background.title = "white", fontcolor.title = "black")

# Plot tracks for candidate enhancer regions of CRT genes only
pdf("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_ZFHX4_candidate_enhancers_Gviz_CRT_genes_only.pdf", height = 3, width = 11)
plotTracks(list(overlapTracks, WTC11_iNeuron_NDD_interaction_track, ATAC_track, H3k27ac, REtrack, TSStrack, biomTrack), 
           stackHeight = 0.3, transcriptAnnotation = "symbol", from = chromStart, to = chromEnd, 
           sizes = c(0.4, 0.15, 0.2, 0.2, 0.2, 0.2, 0.2), labelPos = "below")
dev.off()


```


```{r}

##Fig. S4a

##GNAS Gviz

##Gviz plot 

# Set genomic coordinates for plotting
symbol = "GNAS"
genome = "hg38"
chrom = "chr20"
chromStart = 58260000
chromEnd = 59480000

# load biomaRt data from ensembl
biomart = biomaRt::useMart(biomart = "ensembl", dataset = "hsapiens_gene_ensembl")

# create an ideogram track to display chromosome
#iTrack <- IdeogramTrack(genome = genome, chromosome = chrom, name = chrom)

# create a genome axis track to display chromosome coordinates
gTrack <- GenomeAxisTrack()

# create track with ensembl genes
biomTrack <- BiomartGeneRegionTrack(genome = genome, chromosome = chrom, name = "ENSEMBL", 
                                    filter = list(with_refseq_mrna=TRUE), biomart = biomart)



### (2) Create plot of candidate enhancers, only show contacts for CRT gene list #############################

# Get the list of CRT candidate genes within the genomic window
CRISPRa_CRT_candidates <- data.frame(read_xlsx(path = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/Table_S1_Cis_Regulation_Therapy_Candidate_Genes.xlsx", sheet = "Candidate_Genes"))
CRISPRa_CRT_candidates$gene_symbol[which(CRISPRa_CRT_candidates$chrom == chrom & CRISPRa_CRT_candidates$start_hg38 > chromStart & CRISPRa_CRT_candidates$end_hg38 < chromEnd)]

# Get all 5,425 candidate enhancers of CRT genes
candidate_enhancers_for_CRT_genes <- read.table(file = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/candidate_enhancer_regions_for_CRISPRa_CRT_genes.bed", header = FALSE, sep = "\t",
                                                col.names = c("chrom", "chromStart", "chromEnd", "name"))

# Filter the genomic window of candidate enhancers for plotting
candidate_enhancers_for_CRT_genes <- candidate_enhancers_for_CRT_genes[which(candidate_enhancers_for_CRT_genes$chrom == chrom & candidate_enhancers_for_CRT_genes$chromStart > chromStart & candidate_enhancers_for_CRT_genes$chromEnd < chromEnd),]

# Create regulatory element annotation track
REtrack <- AnnotationTrack(start = candidate_enhancers_for_CRT_genes$chromStart, 
                           end = candidate_enhancers_for_CRT_genes$chromEnd, 
                           chromosome = chrom, genome = genome, stacking = "dense",
                           name = "Candidate Regulatory Elements") 


# Get all 761 TSSs of CRT genes
TSSs_for_CRT_genes <- read.table(file = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/CRISPRa_CRT_Candidates_gencode.v38.basic.coding.transcripts.500bp_promoters.bed", header = FALSE, sep = "\t",
                                                col.names = c("chrom", "chromStart", "chromEnd", "name", "number", "strand")) %>% 
  type_convert()

# Filter the genomic window of candidate enhancers for plotting
TSSs_for_CRT_genes <- TSSs_for_CRT_genes[which(TSSs_for_CRT_genes$chrom == chrom & TSSs_for_CRT_genes$chromStart > chromStart & TSSs_for_CRT_genes$chromEnd < chromEnd),]

# Create TSS annotation track
TSStrack <- AnnotationTrack(start = TSSs_for_CRT_genes$chromStart, 
                           end = TSSs_for_CRT_genes$chromEnd, 
                           chromosome = chrom, genome = genome, stacking = "dense",
                           name = "Transcription_Start_Sites") 



# Create genomic interaction dataframe for each candidate enhancer dataset
GW18_PFC_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/gw18_pfc_ABC_Score_liftOver_hg38_NDD.bedpe",
                                                type = "bedpe", experiment_name = "GW18_PFC", description = "ABC_Score")
WTC11_iNeuron_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/NGN2-neuron_ABC_Score_hg38_MPRA_CRISPRa_validated_for_Gviz.bedpe",
                                                     type = "bedpe", experiment_name = "WTC11_iNeuron", description = "ABC_Score")
Cicero_Fetal_Cerebrum_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/midfetal_cerebrum_ExN_Cicero_liftOver_hg38_NDD.bedpe",
                                                             type = "bedpe", experiment_name = "Cicero_Fetal_Cerebrum", description = "Cicero")
Trevino_Fetal_Cortex_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/trevino_fetal_cortex_enhancer_predictions_NDD.bedpe",
                                                            type = "bedpe", experiment_name = "Trevino_Fetal_Cortex", description = "Correlation")
Ziffra_Fetal_Cortex_NDD <- makeGenomicInteractionsFromFile(fn = "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/ziffra_fetal_human_cortex_ABC_Score_NDD_for_gviz_figures_only.bedpe",
                                                           type = "bedpe", experiment_name = "Ziffra_Fetal_Cortex", description = "ABC_Score")

# only plot positive scores for Trevino
Trevino_Fetal_Cortex_NDD <- Trevino_Fetal_Cortex_NDD[which(Trevino_Fetal_Cortex_NDD@elementMetadata$counts > 0),]

# the height of the interaction plot represents the score from each prioritization method
summary(GW18_PFC_NDD@elementMetadata$counts)
summary(WTC11_iNeuron_NDD@elementMetadata$counts)
summary(Cicero_Fetal_Cerebrum_NDD@elementMetadata$counts)
summary(Trevino_Fetal_Cortex_NDD@elementMetadata$counts)
summary(Ziffra_Fetal_Cortex_NDD@elementMetadata$counts)

# Create interaction tracks for each candidate enhancer dataset
GW18_PFC_NDD_interaction_track <- InteractionTrack(GW18_PFC_NDD, name = "GW18_PFC", chromosome = chrom)
WTC11_iNeuron_NDD_interaction_track <- InteractionTrack(WTC11_iNeuron_NDD, name = "WTC11_iNeuron", chromosome = chrom)
Cicero_Fetal_Cerebrum_NDD_interaction_track <- InteractionTrack(Cicero_Fetal_Cerebrum_NDD, name = "Cicero_Fetal_Cerebrum", chromosome = chrom)
Trevino_Fetal_Cortex_NDD_interaction_track <- InteractionTrack(Trevino_Fetal_Cortex_NDD, name = "Trevino_Fetal_Cortex", chromosome = chrom)
Ziffra_Fetal_Cortex_NDD_interaction_track <- InteractionTrack(Ziffra_Fetal_Cortex_NDD, name = "Ziffra_Fetal_Cortex", chromosome = chrom)

# Show the available display parameters for a Gviz track
# availableDisplayPars(gTrack)


##Adding an ENCODE ATAC track


bamFile <- "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/WTC11_NGN2_iPSC_7-8wk_ExN_ATAC-seq_Song_2019_hg38.50bp_trim.merged.srt.nodup.no_chrM_MT.filtered.bw"
ATAC_track <- DataTrack(range = bamFile, genome = genome, type = "l", 
                     name = "ATAC", window = -1, 
                     chromosome = chrom, type = "hist", 
                    col.histogram = "black", 
                    fill.histogram = "black", background.title = "white", 
                    fontcolor.title = "black", ylim = c(0,5))
class(ATAC_track)

ATAC_track



##Adding a H3k27Ac track


bamFile <- "/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/WTC11_NGN2_iPSC_7-8wk_ExN_H3K27ac_CUT+RUN_Song_2019_hg38.srt.no_dups.filtered.bw"
H3k27ac <- DataTrack(range = bamFile, genome = genome, type = "l", 
                     name = "H3K27AC", window = -1, 
                     chromosome = chrom, type = "hist", 
                    col.histogram = "black", 
                    fill.histogram = "black", background.title = "white", 
                    fontcolor.title = "black", ylim = c(0,10))
class(H3k27ac)

H3k27ac


##Then reading in the gRNA mappings and function data and generating the gRNA function score track

gRNA_Coordinates <- read_tsv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/Scaled_Screen_gRNA_coordinates.txt") %>% 
  dplyr::rename(Chromosome = seqname, gRNA_Name = patternID) %>% 
  separate(Chromosome, into = c("Chr", "Chromosome"),
            sep= 3) %>%
  arrange(gRNA_Name) 

gRNA_Coordinates <- gRNA_Coordinates %>% 
  separate(gRNA_Name, into = c("gRNA_ID", "gRNA_Name"), sep = "_", remove = FALSE) %>% 
  type_convert()

gRNA_DE_Results <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Results.csv") 

gRNA_DE_Results <- gRNA_DE_Results %>% 
  left_join(gRNA_Coordinates, by = "gRNA_ID")

##convert ensg ids to hgnc symbols and filter

mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- gRNA_DE_Results$response_id
G_list <- getBM(filters= "ensembl_gene_id", attributes= c("hgnc_symbol", "ensembl_gene_id"), values=genes, mart= mart)

G_list <- G_list %>% 
  dplyr::rename(response_id = ensembl_gene_id) %>% 
  dplyr::rename(target_gene = hgnc_symbol)

gRNA_DE_Results <- gRNA_DE_Results %>% 
  left_join(G_list, by = "response_id")

gRNA_Data <- gRNA_DE_Results %>% 
  filter(target_gene == symbol) %>% 
  unite(gRNA_Gene_Pair, gRNA_ID, target_gene, remove = FALSE) 

#gRNA_Data <- gRNA_Data %>%
  #distinct(gRNA_Gene_Pair, .keep_all = TRUE)

gRNA_Data <- gRNA_Data %>%
  mutate(plot_pval = -log10(p_value))

##Filtering for hits or non-hits and coloring 

PT_Hits <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Hits_Annotated.csv")

Promoter_Hits <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Hits_Annotated.csv") %>% filter(Target_CRE_Class == "Promoter")

Enhancer_Hits <- read_csv("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Data/Figure_Data_IS/SCEPTRE_PT_Hits_Annotated.csv") %>% filter(Target_CRE_Class == "Enhancer")

Promoter_Hits <- gRNA_Data %>% 
  filter(gRNA_ID %in% Promoter_Hits$gRNA_ID)
Promoter_Hits$Hit <- "Hit"

Enhancer_Hits <- gRNA_Data %>% 
  filter(gRNA_ID %in% Enhancer_Hits$gRNA_ID)
Enhancer_Hits$Hit <- "Hit"

Non_Hits <- gRNA_Data %>% 
  filter(!gRNA_ID %in% PT_Hits$gRNA_ID)
Non_Hits$Hit <- "Non-Hit" 

##Create hit and non-hit tracks and overlay them

df1_label = "Non-Hit"
df2_label = "Promoter_Hit"
df3_label = "Enhancer_Hit"
df1 <- Non_Hits
df2 <- Promoter_Hits
df3 <- Enhancer_Hits

Enhancer_Hit_Track <- DataTrack(data = Enhancer_Hits$plot_pval, start = Enhancer_Hits$start, end = Enhancer_Hits$end, chromosome = chrom, genome = genome, groups = factor(df3_label,levels = c(df1_label, df2_label, df3_label)), col = "#56B4E9", cex = 1.2, legend=FALSE, ylim = c(0,14), col.axis = "black", background = "white", fontcolor.title = "black", name = "-log10(P-values)", background.title = "white", col.frame = "white", fill = "white")

Promoter_Hit_Track <- DataTrack(data = Promoter_Hits$plot_pval, start = Promoter_Hits$start, end = Promoter_Hits$end, chromosome = chrom, genome = genome, groups = factor(df2_label,levels = c(df1_label, df2_label, df3_label)), col = "#E69F00", cex = 1.2, legend=FALSE, ylim = c(0,14), col.axis = "black", background = "white", fontcolor.title = "black", name = "-log10(P-values)", background.title = "white", col.frame = "white", fill = "white")

Non_Hit_Track <- DataTrack(data = Non_Hits$plot_pval, start = Non_Hits$start, end = Non_Hits$end, chromosome = chrom, genome = genome, groups = factor(df1_label,levels = c(df1_label, df2_label)), col = "#999999", legend=FALSE, ylim = c(0,14), col.axis = "black", background = "white", fontcolor.title = "black", name = "-log10(P-values)", background.title = "white", col.frame = "white", fill = "white")

overlapTracks <- OverlayTrack(trackList = list(Non_Hit_Track, Promoter_Hit_Track, Enhancer_Hit_Track), legend= TRUE, background.title = "white", col.frame = "white", fill = "white")



# Set track parameters for plotting
displayPars(gTrack) <- list(fontsize = 6)
displayPars(GW18_PFC_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(WTC11_iNeuron_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(Cicero_Fetal_Cerebrum_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(Trevino_Fetal_Cortex_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(Ziffra_Fetal_Cortex_NDD_interaction_track) = list(col.interactions="#56B4E9", col.anchors.fill ="white", col.anchors.line = "white", interaction.dimension = "height", interaction.measure = "counts", plot.trans = FALSE, plot.outside = TRUE, col.outside = "#56B4E9", anchor.height = 0, background.title = "white", fontcolor.title = "black")
displayPars(REtrack) <- list(col = "#56B4E9", fill = "#56B4E9", col.axis = "black", background.title = "white", fontcolor.title = "black")
displayPars(TSStrack) <- list(col = "#E69F00", fill = "#E69F00", col.axis = "black", background.title = "white", fontcolor.title = "black")
displayPars(biomTrack) <- list(background.title = "white", fontcolor.title = "black", col = "#999999", fill = "#999999", col.line = "#999999", utr3 = "#999999", utr5 = "#999999", protein_coding = "#999999")
displayPars(ATAC_track) <- list(background.title = "white", fontcolor.title = "black")
displayPars(H3k27ac) <- list(background.title = "white", fontcolor.title = "black")

# Plot tracks for candidate enhancer regions of CRT genes only
pdf("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_GNAS_candidate_enhancers_Gviz_CRT_genes_only.pdf", height = 3, width = 11)
plotTracks(list(overlapTracks, WTC11_iNeuron_NDD_interaction_track, ATAC_track, H3k27ac, REtrack, TSStrack, biomTrack), 
           stackHeight = 0.3, transcriptAnnotation = "symbol", from = chromStart, to = chromEnd, 
           sizes = c(0.4, 0.15, 0.2, 0.2, 0.2, 0.2, 0.4), labelPos = "below")
dev.off()




```

```{r}
##Fig. S4a

set.seed(10)

##GNAS Box plots

##GNAS gRNA #1

##Defining gRNA & Target gene

gRNA <- "13174_GNAS_Promoter_Horlbeck_etal"
Target_Gene <- "ENSG00000087460"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#E69F00")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_13174_GNAS_Boxplot.jpeg", width = 3.75, height = 4.75)


##GNAS gRNA #2

##Defining gRNA & Target gene

gRNA <- "13175_GNAS_Promoter_Horlbeck_etal"
Target_Gene <- "ENSG00000087460"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#E69F00")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_13175_GNAS_Boxplot.jpeg", width = 3.75, height = 4.75)

##GNAS gRNA #3

##Defining gRNA & Target gene

gRNA <- "13176_GNAS_Promoter_Horlbeck_etal"
Target_Gene <- "ENSG00000087460"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#E69F00")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_13176_GNAS_Boxplot.jpeg", width = 3.75, height = 4.75)

##GNAS gRNA #4

##Defining gRNA & Target gene

gRNA <- "4444_chr20:59231756-59232689()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000087460"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_4444_GNAS_Boxplot.jpeg", width = 3.75, height = 4.75)

##GNAS gRNA #5

##Defining gRNA & Target gene

gRNA <- "4443_chr20:59231756-59232689()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000087460"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_4443_GNAS_Boxplot.jpeg", width = 3.75, height = 4.75)

##GNAS gRNA #6

##Defining gRNA & Target gene

gRNA <- "4453_chr20:59398520-59399020()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000087460"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_4453_GNAS_Boxplot.jpeg", width = 3.75, height = 4.75)

##GNAS gRNA #7

##Defining gRNA & Target gene

gRNA <- "4457_chr20:59461341-59461841()_Enhancer_Flashfry"
Target_Gene <- "ENSG00000087460"

##Isolating cells with the gRNA

gRNA_Cells <- as.data.frame(gRNA_Matrix[gRNA, ]) 

gRNA_Cells <- gRNA_Cells %>% 
  dplyr::rename("gRNA_Cells" = 1) %>% 
  filter(gRNA_Cells > 5)

rownames(gRNA_Cells)

Gene_Matrix_gRNA_Cells <- as.data.frame(Gene_Matrix[, rownames(gRNA_Cells)]) 

Gene_Matrix_gRNA_Cells <- rownames_to_column(Gene_Matrix_gRNA_Cells, "Ensembl_ID")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Gene_Matrix_gRNA_Cells_2 <- Gene_Matrix_gRNA_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Gene_Matrix_gRNA_Cells_2$Group <- "gRNA_Cells"

Gene_Matrix_gRNA_Cells_2_Target_Gene <- Gene_Matrix_gRNA_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Isolating control cells

Control_Cells <- as.data.frame(Gene_Matrix[,sample(ncol(Gene_Matrix), size = 2*length(gRNA_Cells$gRNA_Cells)), drop = FALSE])

Control_Cells <- rownames_to_column(Control_Cells, "Ensembl_ID")

Control_Cells_2 <- Control_Cells %>% 
  dplyr::select(!any_of(rownames(gRNA_Cells))) %>% 
  dplyr::select(1:(length(gRNA_Cells$gRNA_Cells)))

Control_Cells_2 <- Control_Cells_2 %>% 
  pivot_longer(!Ensembl_ID, names_to = "Cell_BC", values_to = "UMI_Count")

Control_Cells_2 <- Control_Cells_2 %>% 
  group_by(Cell_BC) %>% 
  mutate(Normalized_Expression = log(((UMI_Count/sum(UMI_Count))*10000)+1))

Control_Cells_2$Group <- "Control_Cells"

Control_Cells_Target_Gene <- Control_Cells_2 %>% 
  filter(Ensembl_ID == Target_Gene) 

##Combining for plotting

gRNA_vs_Control_Cells_Target_Gene <- rbind(Gene_Matrix_gRNA_Cells_2_Target_Gene, Control_Cells_Target_Gene)


ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=UMI_Count)) + 
  geom_boxplot() +  
  geom_jitter(height = 0, width = 0.2) +
  theme_classic() 

ggplot(gRNA_vs_Control_Cells_Target_Gene, aes(x=Group, y=Normalized_Expression, colour = Group, fill = Group)) + 
  theme_classic() +
  scale_fill_manual(values=c("#999999", "#56B4E9")) +
  geom_boxplot(colour = "black", width = 0.5, lwd =0.6, outlier.shape = NA) +
  geom_jitter(size = 0.75, colour = "black", width = 0.15, height = 0) +
  theme(axis.line = element_line(colour = 'black', size = 0.6)) +
  theme(axis.ticks = element_line(colour = "black", size = 0.6)) +
  theme(axis.ticks.length=unit(.2, "cm"), legend.position = "none") +
  theme(legend.position = "none") +
  theme(text = element_text(family="Arial", colour = "black", size = 33)) +
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
  labs(title = "", x = "", y = "") 
ggsave("/Users/troymcdiarmid/Documents/Neurohub/Scaled_Screen/Figs/Pub_Figs/Figure_Embedded_Data_Visualizations/Fig_4_gRNA_4457_GNAS_Boxplot.jpeg", width = 3.75, height = 4.75)

```

