# Large-scale discovery of neural enhancers for cis-regulation therapies 

Troy A. McDiarmid1,2,†, Nicholas F. Page3,4,5,†, Florence M. Chardon1,2,†, Riza M. Daza1,2, George T. Chen6,7, Michael Kosicki8 , Lucas M. James9,10, Hannah C. Nourie9,10, Dianne Laboy-Cintrón4,5, Arthur S. Lee11,12,13, Paula Vij14, Diego Calderon1, Jean-Benoît Lalanne1, Beth K. Martin1,2, Kyle Fink14, Michael E. Talkowski11,12,13, Alysson R. Muotri15,16, Benjamin D. Philpot9,10,17, Len A. Pennacchio8,18,19, Daniel H. Geschwind6,7,20,21,22, Stephan J. Sanders3,5,23, Nadav Ahituv4,5*, and Jay Shendure1,2,24,25,26*

Affiliations:
1. Department of Genome Sciences, University of Washington, Seattle, WA, USA
2. Seattle Hub for Synthetic Biology, Seattle, WA, USA
3. Department of Psychiatry and Behavioral Sciences, University of California, San Francisco, San Francisco, CA, USA
4. Department of Bioengineering and Therapeutic Sciences, University of California, San Francisco, San Francisco, CA, USA
5. Institute for Human Genetics, University of California, San Francisco, San Francisco, CA, USA
6. Program in Neurogenetics, Department of Neurology, David Geffen School of Medicine, UCLA, Los Angeles, CA, USA
7. Center for Autism Research and Treatment, Semel Institute, UCLA, Los Angeles, CA, USA
8. Environmental Genomics & System Biology Division, Lawrence Berkeley National Laboratory, Berkeley, CA, USA
9. UNC Neuroscience Center, The University of North Carolina at Chapel Hill, Chapel Hill, NC, USA
10. Department of Cell Biology and Physiology, The University of North Carolina at Chapel Hill, Chapel Hill, NC, USA
11. Center for Genomic Medicine, Massachusetts General Hospital, Boston, MA, USA
12. Program in Medical & Population Genetics & Stanley Center for Psychiatric Research, Broad Institute of MIT & Harvard, Cambridge, MA
13. Department of Neurology, Harvard Medical School, Boston, MA
14. Stem Cell Program and Gene Therapy Center, Institute for Regenerative Cures, MIND Institute, and Department of Neurology, University of California Davis Health, Sacramento, CA 95817
15. Departments of Pediatrics; School of Medicine, UCSD, La Jolla, CA 92093, USA
16. Departments of Cellular Molecular Medicine; School of Medicine, UCSD, La Jolla, CA 92093, USA
17. Carolina Institute for Developmental Disabilities, The University of North Carolina at Chapel Hill, Chapel Hill, NC, USA
18. U.S. Department of Energy Joint Genome Institute, One Cyclotron Road, Berkeley, CA, USA
19. Comparative Biochemistry Program, University of California, Berkeley, CA, USA
20. Department of Psychiatry and Behavioral Sciences; David Geffen School of Medicine, UCLA, Los Angeles, CA, USA
21. Department of Human Genetics, David Geffen School of Medicine, UCLA, Los Angeles, CA, USA
22. Institute of Precision Health, David Geffen School of Medicine, UCLA, Los Angeles, CA, USA
23. Institute of Developmental and Regenerative Medicine, Department of Paediatrics, University of Oxford, Oxford, UK
24. Brotman Baty Institute for Precision Medicine, Seattle, WA, USA
25. Howard Hughes Medical Institute, Seattle, WA, USA
26. Allen Discovery Center for Cell Lineage Tracing, Seattle, WA, USA

†. These authors contributed equally to this work
          
* Correspondence to N.A. (Nadav.Ahituv@ucsf.edu) or J.S. (shendure@uw.edu)

Abstract
CRISPR-based gene activation (CRISPRa) has emerged as a promising therapeutic approach for neurodevelopmental disorders (NDD) caused by haploinsufficiency. However, scaling this cis-regulatory therapy (CRT) paradigm requires pinpointing which candidate cis-regulatory elements (cCREs) are active in human neurons, and which can be targeted with CRISPRa to yield specific and therapeutic levels of target gene upregulation. Here, we combine Massively Parallel Reporter Assays (MPRAs) and a multiplex single cell CRISPRa screen to discover functional human neural enhancers whose CRISPRa targeting yields specific upregulation of NDD risk genes. First, we tested 5,425 candidate neuronal enhancers with MPRA, identifying 2,422 that are active in human neurons. Selected cCREs also displayed specific, autonomous in vivo activity in the developing mouse central nervous system. Next, we applied multiplex single-cell CRISPRa screening with 15,643 gRNAs to test all MPRA-prioritized cCREs and 761 promoters of NDD genes in their endogenous genomic contexts. We identified hundreds of promoter- and enhancer-targeting CRISPRa gRNAs that upregulated 200 of the 337 NDD genes in human neurons, including 91 novel enhancer-gene pairs. Finally, we confirmed that several of the CRISPRa gRNAs identified here demonstrated selective and therapeutically relevant upregulation of SCN2A, CHD8, CTNND2 and TCF4 when delivered virally to patient cell lines, human cerebral organoids, and a humanized mouse model of hTcf4. Our results provide a comprehensive resource of active, target-linked human neural enhancers for NDD genes and corresponding gRNA reagents for CRT development. More broadly, this work advances understanding of neural gene regulation and establishes a generalizable strategy for discovering CRT gRNA candidates across cell types and haploinsufficient disorders.





# Repository Contents


This repository contains design, analysis, and visualization code for "Large-scale discovery of neural enhancers for cis-regulation therapies". The visualization scripts labeled "Fig_1_Viz.Rmd", "Fig_2_Viz.Rmd" etc. were used to create all visualizations in the manuscript. Raw sequencing data, custom sequencing amplicons, and processed data files generated in this study have been deposited on the IGVF portal and are freely available with accession numbers IGVFDS1120GRVG (MPRA) and IGVFDS2290SSEF (single cell CRISPRa screen). 
